astropy/cosmology/_src/traits/scale_factor.py

```diff
@@ -0,0 +1,56 @@
+"""Scale factor.
+
+This is private API. See `~astropy.cosmology.parts` for public API.
+
+"""
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+__all__ = ["ScaleFactor"]
+
+
+from numpy.typing import ArrayLike
+
+import astropy.units as u
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+
+
+class ScaleFactor:
+    """The object has attributes and methods for computing the cosmological scale factor.
+
+    The scale factor is defined as :math:`a = a_0 / (1 + z)`.
+
```

Comment:
Do you need the empty line?

Reply:
All the numpydoc https://numpydoc.readthedocs.io/en/latest/format.html has it. 🤷